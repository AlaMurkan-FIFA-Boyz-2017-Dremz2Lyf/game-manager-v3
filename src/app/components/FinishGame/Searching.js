import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Segment, Message, Loader } from 'semantic-ui-react';

class Searching extends PureComponent {
  static propTypes = {
    loading: PropTypes.bool,
    fetchImages: PropTypes.func,
  };

  componentDidMount() {
    this.props.fetchImages();
  }

  render() {
    const { loading } = this.props;

    return (
      <Segment basic>
        <Message color="teal">
          <Message.Header>
            Searching Twitter for your results.
          </Message.Header>
        </Message>
        {loading ? <Loader size="huge" active={loading} inline="centered" /> : null}
      </Segment>
    );
  }
}

export default Searching;
