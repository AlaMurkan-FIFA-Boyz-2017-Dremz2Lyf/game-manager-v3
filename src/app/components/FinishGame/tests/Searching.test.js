import React from 'react';
import { shallow } from 'enzyme';

import Searching from '../Searching';

describe('Searching', () => {
  let wrapper;
  let fetchImagesMock;

  beforeEach(() => {
    fetchImagesMock = jest.fn();
  });

  afterEach(() => {
    fetchImagesMock.mockReset();
  });

  it('should match the snapshot when loading', () => {
    wrapper = shallow(
      <Searching loading fetchImages={fetchImagesMock} />,
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Loader').length).toBe(1);
    expect(fetchImagesMock).toHaveBeenCalled();
  });

  it('should match the snapshot when loaded', () => {
    wrapper = shallow(
      <Searching loading={false} fetchImages={fetchImagesMock} />,
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Loader').length).toBe(0);
    expect(fetchImagesMock).toHaveBeenCalled();
  });
});
