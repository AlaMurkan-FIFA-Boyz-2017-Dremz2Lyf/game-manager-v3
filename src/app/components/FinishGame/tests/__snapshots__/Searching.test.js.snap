// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searching should match the snapshot when loaded 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Searching
    fetchImages={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    loading={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "basic": true,
      "children": Array [
        <Message
          color="teal"
        >
          <MessageHeader>
            Searching Twitter for your results.
          </MessageHeader>
        </Message>,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <MessageHeader>
            Searching Twitter for your results.
          </MessageHeader>,
          "color": "teal",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Searching Twitter for your results.",
          },
          "ref": null,
          "rendered": "Searching Twitter for your results.",
          "type": [Function],
        },
        "type": [Function],
      },
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "basic": true,
        "children": Array [
          <Message
            color="teal"
          >
            <MessageHeader>
              Searching Twitter for your results.
            </MessageHeader>
          </Message>,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <MessageHeader>
              Searching Twitter for your results.
            </MessageHeader>,
            "color": "teal",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Searching Twitter for your results.",
            },
            "ref": null,
            "rendered": "Searching Twitter for your results.",
            "type": [Function],
          },
          "type": [Function],
        },
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Searching should match the snapshot when loading 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Searching
    fetchImages={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    loading={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "basic": true,
      "children": Array [
        <Message
          color="teal"
        >
          <MessageHeader>
            Searching Twitter for your results.
          </MessageHeader>
        </Message>,
        <Loader
          active={true}
          inline="centered"
          size="huge"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <MessageHeader>
            Searching Twitter for your results.
          </MessageHeader>,
          "color": "teal",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Searching Twitter for your results.",
          },
          "ref": null,
          "rendered": "Searching Twitter for your results.",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": true,
          "inline": "centered",
          "size": "huge",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "basic": true,
        "children": Array [
          <Message
            color="teal"
          >
            <MessageHeader>
              Searching Twitter for your results.
            </MessageHeader>
          </Message>,
          <Loader
            active={true}
            inline="centered"
            size="huge"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <MessageHeader>
              Searching Twitter for your results.
            </MessageHeader>,
            "color": "teal",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Searching Twitter for your results.",
            },
            "ref": null,
            "rendered": "Searching Twitter for your results.",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": true,
            "inline": "centered",
            "size": "huge",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
